features/rerun-only.feature

```diff
@@ -0,0 +1,180 @@
+          Then I should have <result> apples
+
+          Examples:
+            | ate | found | result |
+            | 3   | 1     | 1      |
+            | 0   | 4     | 8      |
+            | 2   | 2     | 3      |
+      """
+
+  Scenario: Run one feature with 2 failed and 3 passing scenarios
+    When I run "behat --no-colors -f progress features/apples.feature"
+    Then it should fail with:
+      """
+      ..F.............F....
+
+      --- Failed steps:
+
+      001 Scenario: I'm little hungry   # features/apples.feature:9
+            Then I should have 3 apples # features/apples.feature:11
+              Failed asserting that 2 matches expected 3.
+
+      002 Example: | 0   | 4     | 8      | # features/apples.feature:29
+            Then I should have 8 apples     # features/apples.feature:24
+              Failed asserting that 7 matches expected 8.
+
+      6 scenarios (4 passed, 2 failed)
+      21 steps (19 passed, 2 failed)
+      """
+
+  Scenario: Rerun only failed scenarios
+    Given I run "behat --no-colors -f progress features/apples.feature"
+    When I run "behat --no-colors -f progress features/apples.feature --rerun-only"
+    Then it should fail with:
+    """
+    ..F...F
+
+    --- Failed steps:
+
+    001 Scenario: I'm little hungry   # features/apples.feature:9
+          Then I should have 3 apples # features/apples.feature:11
+            Failed asserting that 2 matches expected 3.
+
+    002 Example: | 0   | 4     | 8      | # features/apples.feature:29
+          Then I should have 8 apples     # features/apples.feature:24
+            Failed asserting that 7 matches expected 8.
+
+    2 scenarios (2 failed)
+    7 steps (5 passed, 2 failed)
+    """
+
+  Scenario: Fixing scenario removes it from the rerun log
+    Given I run "behat --no-colors -f progress features/apples.feature"
+    And there is a file named "features/apples.feature" with:
+      """
+      Feature: Apples story
+        In order to eat apple
+        As a little kid
+        I need to have an apple in my pocket
+
+        Background:
+          Given I have 3 apples
+
+        Scenario: I'm little hungry
+          When I ate 1 apple
+          Then I should have 2 apples
+
+        Scenario: Found more apples
+          When I found 5 apples
+          Then I should have 8 apples
+
+        Scenario: Found more apples
+          When I found 2 apples
+          Then I should have 5 apples
+
+        Scenario Outline: Other situations
+          When I ate <ate> apples
+          And I found <found> apples
+          Then I should have <result> apples
+
+          Examples:
+            | ate | found | result |
+            | 3   | 1     | 1      |
+            | 0   | 4     | 7      |
+            | 2   | 2     | 3      |
+      """
+    When I run "behat --no-colors -f progress features/apples.feature"
+    And I run "behat --no-colors -f progress features/apples.feature --rerun-only"
+    Then it should pass with:
+    """
```

Comment:
If we added this feature, I think it should render an explicit message to output - otherwise it could be difficult to distinguish from an unexpected exit.